lms/djangoapps/courseware/module_render.py

```diff
@@ -462,17 +474,16 @@ def inner_get_module(descriptor):
         )
 
     def publish(block, event_type, event):
-        """A function that allows XModules to publish events."""
-        if event_type == 'grade' and not is_masquerading_as_specific_student(user, course_id):
-            SCORE_PUBLISHED.send(
-                sender=None,
-                block=block,
-                user=user,
-                raw_earned=event['value'],
-                raw_possible=event['max_value'],
-                only_if_higher=event.get('only_if_higher'),
-                score_deleted=event.get('score_deleted'),
-            )
+        """
+        A function that allows XModules to publish events.
+        """
+        special_events = {
+            'completion': handle_completion_event,
+            'grade': handle_grade_event,
+            'progress': handle_progress_event,
+        }
+        if event_type in special_events:
+            special_events[event_type](block, event)
```

Comment:
1. Could we check here for `is_masquerading_as_specific_student(user, course_id)` if it is a reasonable default check for future events to ensure it is not forgotten?  Would it hurt to have that check for 'progress' events?
2. In all cases, is there a test for the user checks?  Do you need one for the special progress handler user check as well?

Reply:
I made the check general.  There could be future events that would need to be run even when masquerading, but that would be an easy refactor later.